src/ducks/transactions/TransactionModal.jsx

```diff
@@ -163,15 +232,42 @@ const TransactionModalInfoContent = withTransaction(props => {
             {
               label: t('Transactions.infos.originalBankLabel'),
               value: flag('originalBankLabel') && transaction.originalBankLabel
+            },
+            {
+              label: t('Transactions.infos.date'),
+              value: transaction.applicationDate
+                ? f(getDate(transaction), 'dddd D MMMM')
+                : null
             }
           ].filter(x => x.value)}
         />
       </TransactionModalRow>
-      <TransactionModalRow iconLeft={iconCalendar}>
-        <span style={transactionModalRowStyle}>
-          {f(getDate(transaction), 'dddd DD MMMM')}
-        </span>
-      </TransactionModalRow>
+      <TransactionModalRowMedia onClick={handleShowApplicationEditor}>
+        <Img>
+          <Icon icon={iconCalendar} />
+        </Img>
+        <Bd>
+          {transaction.applicationDate
+            ? t('Transactions.infos.applicationDate', {
+                applicationDate: f(transaction.applicationDate, 'MMMM YYYY')
+              })
+            : f(getDate(transaction), 'dddd D MMMM')}
+        </Bd>
+        {transaction.applicationDate && !applicationDateBusy ? (
```

Comment:
Nit: This condition could be extracted to a `shouldShowRestoreIcon` variable. Render functions are easier to read with this kind of variables instead of inlined conditions.